jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/utils/JMSUtils.java

```diff
@@ -0,0 +1,180 @@
+     *
+     * @param message JMS message that need to be changed as carbon message
+     * @return the carbon message converted from jms message
+     */
+    public static CarbonMessage createJMSCarbonMessage(Message message) throws JMSConnectorException {
+        CarbonMessage jmsCarbonMessage = null;
+        try {
+            if (message instanceof TextMessage) {
+                jmsCarbonMessage = new TextCarbonMessage(((TextMessage) message).getText());
+                jmsCarbonMessage.setProperty(JMSConstants.JMS_MESSAGE_TYPE, JMSConstants.TEXT_MESSAGE_TYPE);
+            } else if (message instanceof MapMessage) {
+                MapCarbonMessage mapCarbonMessage = new MapCarbonMessage();
+                MapMessage mapMessage = (MapMessage) message;
+                Enumeration<String> mapKeys = ((MapMessage) message).getMapNames();
+                while (mapKeys.hasMoreElements()) {
+                    String mapKey = mapKeys.nextElement();
+                    String mapValue = mapMessage.getString(mapKey);
+                    if (null != mapValue) {
+                        mapCarbonMessage.setValue(mapKey, mapValue);
+                    }
+                }
+                jmsCarbonMessage = mapCarbonMessage;
+                jmsCarbonMessage.setProperty(JMSConstants.JMS_MESSAGE_TYPE, JMSConstants.MAP_MESSAGE_TYPE);
+            } else if (message instanceof ObjectMessage) {
+                jmsCarbonMessage = (SerializableCarbonMessage) ((ObjectMessage) message).getObject();
+                jmsCarbonMessage.setProperty(JMSConstants.JMS_MESSAGE_TYPE, JMSConstants.OBJECT_MESSAGE_TYPE);
+            } else {
+                jmsCarbonMessage = new TextCarbonMessage(((BytesMessage) message).readUTF());
+                jmsCarbonMessage.setProperty(JMSConstants.JMS_MESSAGE_TYPE, JMSConstants.BYTES_MESSAGE_TYPE);
+            }
+            String messageId = message.getJMSMessageID();
+            if (null != messageId) {
+                jmsCarbonMessage.setHeader(JMSConstants.JMS_MESSAGE_ID, messageId);
+            }
+            jmsCarbonMessage.setHeader(JMSConstants.JMS_DELIVERY_MODE, String.valueOf(message.getJMSDeliveryMode()));
+            jmsCarbonMessage.setHeader(JMSConstants.JMS_PRIORITY, String.valueOf(message.getJMSPriority()));
+            jmsCarbonMessage.setHeader(JMSConstants.JMS_RE_DELIVERED, String.valueOf(message.getJMSRedelivered()));
+            jmsCarbonMessage.setHeader(JMSConstants.JMS_TIME_STAMP, String.valueOf(message.getJMSTimestamp()));
+
+            Enumeration<String> properties = message.getPropertyNames();
+            while (properties.hasMoreElements()) {
+                String name = properties.nextElement();
+                jmsCarbonMessage.setHeader(name, message.getStringProperty(name));
+            }
+            return jmsCarbonMessage;
+        } catch (JMSException e) {
+            logger.error("Error while changing the jms message to carbon message");
```

Comment:
Don't log and throw the error. This is not the best practice, either you handle it or throw it.